NoTargets: unable to be a dependant project "missing assembly"

The NoTargets SDK does not emit an assembly, but still advertises that it outputs an assembly to other projects that take dependencies on it.

### Version information
Component|Version


### Reproduction steps
1. Create three class library projects
```shell
dotnet new classlib -n project1
dotnet new classlib -n project2
dotnet new classlib -n project3
```

2. Add references such that `project3 -> project2 -> project1`
```shell
dotnet add project2 reference project1
dotnet add project3 reference project2
```

3. Edit `project2` and `project3` to use the `Microsoft.Build.NoTargets` SDK (version 1.0.80)
```diff
-<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.Build.NoTargets/1.0.80">
```

4. Build `project3`
```shell
dotnet build project3
```

### Expected result
The build succeeds and the `$(OutputPath)` for `project3` contains the binaries produced for `project1`: `project1.dll`.

### Actual result
The build fails to find `project2.dll`

```
Microsoft (R) Build Engine version 16.3.0-preview-19377-01+dd8019d9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 23.05 ms for /Users/foo/test/project1/project1.csproj.
  Restore completed in 95.78 ms for /Users/foo/test/project2/project2.csproj.
  Restore completed in 95.78 ms for /Users/foo/test/project3/project3.csproj.
  project1 -> /Users/foo/test/project1/bin/Debug/netstandard2.0/project1.dll
  project2 -> 
/usr/local/share/dotnet/sdk/3.0.100-preview8-013656/Microsoft.Common.CurrentVersion.targets(4377,5): error MSB3030: Could not copy the file "/Users/foo/test/project2/bin/Debug/netstandard2.0/project2.dll" because it was not found. [/Users/foo/test/project3/project3.csproj]

Build FAILED.

/usr/local/share/dotnet/sdk/3.0.100-preview8-013656/Microsoft.Common.CurrentVersion.targets(4377,5): error MSB3030: Could not copy the file "/Users/foo/test/project2/bin/Debug/netstandard2.0/project2.dll" because it was not found. [/Users/foo/test/project3/project3.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:02.33
```

### Scenario
The motivation for this setup is for signing and packaging scenarios.

`project1` represents the actual product binaries, and in reality would be several projects (`project1{a,b,c...}`).

`project2` pulls together all the various projects and assemblies that form the overall product or payload. A custom target runs `AfterBuild` to sign all the assemblies.

`project3` builds an installer. A custom target runs `AfterBuild` to take the signed binaries from `project2` and bundles them into an installer, which is then also signed.

--- a/Packages.props
+++ b/Packages.props
@@ -14,7 +14,7 @@
     <PackageReference Update="Microsoft.Build.Framework" Version="16.0.461" />
     <PackageReference Update="Microsoft.Build.Utilities.Core" Version="16.0.461" />
     <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.1.0" />
-    <PackageReference Update="MSBuild.ProjectCreation" Version="1.2.11" />
+    <PackageReference Update="MSBuild.ProjectCreation" Version="1.2.15" />
     <PackageReference Update="Shouldly" Version="3.0.2" />
     <PackageReference Update="xunit" Version="2.4.1" />
     <PackageReference Update="xunit.runner.visualstudio" Version="2.4.1" />

--- a/src/NoTargets.UnitTests/CustomProjectCreatorTemplates.cs
+++ b/src/NoTargets.UnitTests/CustomProjectCreatorTemplates.cs
@@ -15,6 +15,7 @@ namespace Microsoft.Build.NoTargets.UnitTests
             this ProjectCreatorTemplates templates,
             Action<ProjectCreator> customAction = null,
             string path = null,
+            string targetFramework = "netstandard2.0",
             string defaultTargets = null,
             string initialTargets = null,
             string sdk = null,
@@ -35,7 +36,7 @@ namespace Microsoft.Build.NoTargets.UnitTests
                     projectCollection,
                     projectFileOptions)
                 .Import(Path.Combine(currentDirectory, "Sdk", "Sdk.props"))
-                .Property("TargetFramework", "netstandard1.0")
+                .Property("TargetFramework", targetFramework)
                 .CustomAction(customAction)
                 .Import(Path.Combine(currentDirectory, "Sdk", "Sdk.targets"));
         }

--- a/src/NoTargets.UnitTests/NoTargetsTests.cs
+++ b/src/NoTargets.UnitTests/NoTargetsTests.cs
@@ -7,6 +7,7 @@ using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities.ProjectCreation;
 using Shouldly;
 using System.Collections.Generic;
+using System.IO;
 using UnitTest.Common;
 using Xunit;
 
@@ -47,6 +48,34 @@ namespace Microsoft.Build.NoTargets.UnitTests
                 .TryGetPropertyValue("IncludeBuildOutput", out string includeBuildOutput);
 
             includeBuildOutput.ShouldBe("false");
+        }
+
+        [Fact]
+        public void ProjectsCanDependOnNoTargetsProjects()
+        {
+            ProjectCreator project1 = ProjectCreator.Templates.LegacyCsproj(
+                    path: Path.Combine(TestRootPath, "project1", "project1.csproj"))
+                .Save();
+
+            ProjectCreator project2 = ProjectCreator.Templates.NoTargetsProject(
+                    path: Path.Combine(TestRootPath, "project2", "project2.csproj"))
+                .Property("DesignTimeBuild", "true")
+                .Property("GenerateDependencyFile", "false")
+                .Target("_GetProjectReferenceTargetFrameworkProperties")
+                .ItemProjectReference(project1)
+                .Save();
+
+            ProjectCreator project3 = ProjectCreator.Templates.NoTargetsProject(
+                    path: Path.Combine(TestRootPath, "project3", "project3.csproj"))
+                .Property("DesignTimeBuild", "true")
+                .Property("GenerateDependencyFile", "false")
+                .ItemProjectReference(project2)
+                .Target("_GetProjectReferenceTargetFrameworkProperties")
+                .Save();
+
+            project3.TryBuild(out bool result, out BuildOutput buildOutput);
+
+            result.ShouldBeTrue(buildOutput.GetConsoleLog());
         }
 
         [Fact]

--- a/src/NoTargets/Sdk/Sdk.targets
+++ b/src/NoTargets/Sdk/Sdk.targets
@@ -59,6 +59,16 @@
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed\Microsoft.Managed.DesignTime.targets" 
           Condition="'$(DebuggerFlavor)' == '' And Exists('$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed\Microsoft.Managed.DesignTime.targets')" />
 
+  <!--
+    NoTargets projects do not build an assembly so dependent projects shouldn't get a path to the target.  Override the GetTargetPath to do nothing.
+  -->
+  <Target Name="GetTargetPath" />
+
+  <!--
+    The GetTargetPathWithTargetPlatformMoniker target uses a BeforeTargets so the only way to disable it is to override it with an empty target.
+  -->
+  <Target Name="GetTargetPathWithTargetPlatformMoniker" />
+
   <Import Project="$(CustomAfterNoTargets)" Condition="'$(CustomAfterNoTargets)' != '' and Exists('$(CustomAfterNoTargets)')" />
   
 </Project>

